#### Hyphotesis Test_1

##### Set the hypothesis

* H0: Test group completion rate (CR) = Control group completion rate (CR)
* H1: Test group completion rate (CR) != Control group completion rate (CR)

* significance level = 0.05

In [83]:
# import necessary library
from scipy import stats

### z-test (for comparing proportions) 

In [84]:
# Completion rates and sample sizes for test and control groups
completion_rate_test = 0.14
completion_rate_control = 0.12
n_test = 176641
n_control = 140482

# Calculate sample proportions
p1 = completion_rate_test
p2 = completion_rate_control

# Calculate pooled proportion
p = (p1 * n_test + p2 * n_control) / (n_test + n_control)

# Compute standard error
SE = np.sqrt(p * (1 - p) * (1 / n_test + 1 / n_control))

# Compute z-statistic
Z = (p1 - p2) / SE

# Calculate p-value
p_value = 2 * (1 - stats.norm.cdf(np.abs(Z)))  # Two-tailed test

# Print results
print("Z-statistic:", Z)
print("p-value:", p_value)


Z-statistic: 16.5740796708499
p-value: 0.0


##### Decision

Since p-value (Z-statistic: 16.5740796708499, p-value: 0.0) is less than the choosen significance level, we reject the null hypothesis. And , conclude that:  Test group completion rate (CR) is not equal to Control group completion rate (CR).

#### Hyphotesis Test_2

##### Set the hypothesis

* H0: All clients in test group average/mean total process time = All clients in control group average/mean total process time
* H1: All clients in test group average/mean total process time != All clients in control group average/mean total process time

* significance level = 0.05

In [126]:
# Load the data samples
#sample1 = pd.read_csv("time_differ1_test.csv")
#sample2 = pd.read_csv("time_differ1_control.csv")
# Change the datatype to timedelta to both group
#sample1['time_diffrence'] = pd.to_timedelta(sample1['time_diffrence'])
#sample2['time_diffrence'] = pd.to_timedelta(sample2['time_diffrence'])

In [134]:
# Cahnge timedelta to integer dataype to test group (We use the .dt.total_seconds() method to convert the timedeltas to total seconds.
# Then, we use .astype(int) to convert the total seconds to integers.)
#sample1['time_diffrence'] = sample1['time_diffrence'].dt.total_seconds().astype(int)

In [136]:
# Cahnge timedelta to integer dataype to control group
#sample2['time_diffrence'] = sample2['time_diffrence'].dt.total_seconds().astype(int)

### Two sample T-test

In [138]:
t_statistic, p_value = stats.ttest_ind(sample1, sample2, equal_var=True)

# Print the t-statistic and p-value
print("t-statistic:", t_statistic)
print("p-value:", p_value)

t-statistic: [-3.51175486]
p-value: [0.00044523]


##### Decision

Since p-value (t-statistic: [-3.51175486] & p-value: [0.00044523]) is less than the choosen significance level (Alpha : 0.05), we reject the null hypothesis. And , conclude that:  All clients in test group average/mean total process time is not equal to All clients in control group average/mean total process time.

#### Hyphotesis Test_3

##### Set the hypothesis

* H0: All clients in test group average/mean total process time = All clients in control group average/mean total process time
* H1: All clients in test group average/mean total process time != All clients in control group average/mean total process time

* significance level = 0.05